### PR TITLE
libsolv rewrite

### PR DESCRIPTION
**Pull request by ignatenkobrain**

### CODE REVIEW

File: repos.cfg.sample [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[DEFAULT]
+basedir = /home/brain/Projects/fedora-modularity/depchase/repos
+
```
Comment (sgallagher) [inline repos.cfg.sample]: Do we have to have locally-copied repodata to run against, or can it work against a URL?

Comment (sgallagher) [inline repos.cfg.sample]: Also, can we have a repos.cfg.sample that works against the public Rawhide repositories?

File: README.md [hunk 2]
```diff
@@ -1,179 +1,48 @@
...
+## Usage
 
 ```
-
-### Get the list of packages required to self-host for one or more packages
-"Self-hosting" in this context means "all of the packages necessary to be able
-to build all the packages listed on the command line, plus recursively any
-packages required to build those BuildRequires as well.
-
+$ depchase -a x86_64 -c repos.cfg resolve [--selfhost] foo --hint bar
```
Comment (sgallagher) [inline README.md]: This section really needs to describe what `repos.cfg` needs to include. Also how to generate the override directories.

Feel free to copy/modify the scripts I used in https://github.com/sgallagher/baseruntime-package-lists such as `recreate_srpm.sh` for this if it makes sense.

File: repos.cfg.sample (hunk 1 above)
Comment (sgallagher) [inline repos.cfg.sample]: The more I think about this, the more I *really* want us to include a config that actually works (or, ideally, configs known to work for all of the releases we're working on). That way we know that our results will be repeatable by someone else.

File: setup.py [hunk 3]
```diff
@@ -1,48 +1,10 @@
...
+from setuptools import setup
+
+setup(name="depchase",
+      version="1",
+      url="https://github.com/fedora-modularity/depchase",
+      license="MIT",
+      author="Igor Gnatenko",
+      author_email="ignatenkobrain@fedoraproject.org",
+      install_requires=["click>=6"],
+      scripts=["depchase"])
```
Comment (sgallagher) [inline setup.py]: Why are you using `scripts` here rather than `console_scripts` with an entry-point? I'm not necessarily refusing it, but doesn't this mean that it wouldn't be compiled into .pyc and .pyo files?

File: repos.cfg.sample (hunk 1 above)
Comment (ignatenkobrain) [inline repos.cfg.sample]: - Locally-copies only
- We can't provide anything like that, only requirement is to have `$path/repodata/repomd.xml`
- If we want to include some config, we need repodata which I would prefer to not include in the repo because it is really huge (~100M for base and override)

File: setup.py (hunk 3 above)
Comment (ignatenkobrain) [inline setup.py]: we don't have module (we have only CLI tool) when console_scripts with entry-points work only with modules.

File: README.md (hunk 2 above)
Comment (ignatenkobrain) [inline README.md]: config can contain as much sections as you wish (which is repoids) and each section should contain `path` element.. Was not sure how to write that down.

Comment (ignatenkobrain) [inline README.md]: About how to generate override repositories... It is completely up to you how you do that. I just download koji builds, rpmbuild -rs SRPMs and createrepo_c... For rawhide bootstrap I also used mergerepo_c, because composed SRPMs had older versions than f27-rebuild repos... So this part is not related to depchase at all

File: repos.cfg.sample (hunk 1 above)
Comment (sgallagher) [inline repos.cfg.sample]: We don't need to include the repodata, but maybe include a helper script to pull down the necessary repodata locally? Anything we can do to reduce the possibility that two people might get different results is important.

Comment (ignatenkobrain) [inline repos.cfg.sample]: I can write such script, but that will go into baseruntime-package-lists, it has nothing to do with depchase

Comment (sgallagher) [inline repos.cfg.sample]: OK, that makes sense.
